clang/lib/AST/ASTDumper.cpp

```diff
@@ -360,3 +360,34 @@ LLVM_DUMP_METHOD void ConceptReference::dump(raw_ostream &OS) const {
   ASTDumper P(OS, Ctx, Ctx.getDiagnostics().getShowColors());
   P.Visit(this);
 }
+
+//===----------------------------------------------------------------------===//
+// TemplateName method implementations
+//===----------------------------------------------------------------------===//
+
+// FIXME: These are using the TemplateArgument dumper.
```

Comment:
I don't think this is a bug, right? Arguably less optimal than it could be.
I think if there is room for improvement the fixme comment should be more detailed

Reply:
Well you are going to hit dump on a TemplateName, and it's going to dump a TemplateArgument of template kind, which is misleading and incorrect, but still helpful in the narrow context this is a debugging aid not actually used in production.